playlistmanager.lua

```diff
@@ -237,6 +238,66 @@ if settings.system=="auto" then
   end
 end
 
+-- auto calculate showamount
+if settings.showamount=='auto' then
+  -- same as draw_playlist() height
+  local h = 360
+  if settings.scale_playlist_by_window then
+    -- mp.set_osd_ass(0, 0, ass.text) default res_y is 288
+    -- https://mpv.io/manual/stable/#command-interface-ass-events
+    -- https://github.com/mpv-player/mpv/blob/e22a2f04833852ce825eb7c1235d9bdaaa9b2397/sub/osd_libass.c#L111-L112
+    -- https://github.com/mpv-player/mpv/blob/e22a2f04833852ce825eb7c1235d9bdaaa9b2397/sub/ass_mp.h#L33
+    h = 288
+  end
+  
+  local playlist_h = h
+  if mp.get_property("osd-align-x") == "left" and mp.get_property("osd-align-y") == "top" then
+    -- both top and bottom with same padding
+    playlist_h = playlist_h - settings.text_padding_y * 2
+  end
+  
+  -- osd-font-size is based on 720p height
+  -- see https://mpv.io/manual/stable/#options-osd-font-size 
+  -- details in https://mpv.io/manual/stable/#options-sub-font-size
+  -- draw_playlist() is based on 360p or 288p height, need some conversion
+  local fs = mp.get_property_native('osd-font-size') * h / 720
+  -- get the ass font size
+  if settings.style_ass_tags ~= nil then
+    local ass_fs_tag = settings.style_ass_tags:match('\\fs%d+')
+    if ass_fs_tag ~= nil then
+      fs = tonumber(ass_fs_tag:match('%d+'))
+    end
+  end
+ 
+  settings.showamount = math.floor(playlist_h / fs)
+  
+  -- exclude the header line
+  if settings.playlist_header ~= "" then
+    settings.showamount = settings.showamount - 1
+    -- probably some newlines (%N or \N) in the header
+    for _ in settings.playlist_header:gmatch('%%N') do
+      settings.showamount = settings.showamount - 1
+    end
+    for _ in settings.playlist_header:gmatch('\\N') do
+      settings.showamount = settings.showamount - 1
+    end
+  end
+  
+  msg.info('auto showamount: ' .. settings.showamount)
+  
+  -- no word wrapping, let long filenames overflow to the right
+  if settings.style_ass_tags == nil or settings.style_ass_tags == '' then
+    settings.style_ass_tags = '{\\q2}'
+  else
+    -- if wrapstyle tag exists, remove it
+    settings.style_ass_tags = settings.style_ass_tags:gsub('\\q%d+','')
+    -- add to end
+    settings.style_ass_tags = settings.style_ass_tags:gsub('}', '\\q2}')
```

Comment:
Why is this necessary?

Reply:
remove original `\q` tag and add `\q2` to end, the \q2 tag is necessary
>{\fs20\q0} --remove--> {\fs20} --add--> {\fs20\q2}
{\fs20} --remove--> {\fs20} --add--> {\fs20\q2}

Reply:
I see there are different settings. I think I'll still change it to be \q2 in default styles, so it's user controlled rather than setting forcing it.